jib-cli/src/test/java/com/google/cloud/tools/jib/cli/jar/JarProcessorsTest.java

```diff
@@ -39,41 +43,53 @@
   private static final String STANDARD = "jar/standard/emptyStandardJar.jar";
   private static final String JAVA_14_JAR = "jar/java14WithModuleInfo.jar";
 
-  @Mock private TempDirectoryProvider mockTemporaryDirectoryProvider;
+  @Mock private CacheDirectories mockCacheDirectories;
+
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
 
   @Test
   public void testFrom_standardExploded() throws IOException, URISyntaxException {
     Path jarPath = Paths.get(Resources.getResource(STANDARD).toURI());
+    Path destination = temporaryFolder.getRoot().toPath();
+    when(mockCacheDirectories.getExplodedJarCache()).thenReturn(destination);
+
     JarProcessor processor =
-        JarProcessors.from(jarPath, mockTemporaryDirectoryProvider, ProcessingMode.exploded);
-    Mockito.verify(mockTemporaryDirectoryProvider).newDirectory();
+        JarProcessors.from(jarPath, mockCacheDirectories, ProcessingMode.exploded);
+
+    verify(mockCacheDirectories).getExplodedJarCache();
```

Comment:
Maybe we can check if `Files.exists(destination.resolve("some-file"))`?

Reply:
Hm I'm not sure if this is possible in this unit test?`JarProcessors` is used to handle all the logic behind creating a `JarProcessor` instance. It doesn't do the unzipping of the jar or creation of layers.

Reply:
Ah, I see.